### PR TITLE
BCDA-3260 Removed HICN Identifer Flag

### PR DESCRIPTION
**Pull request by PatKelsh**
### Fixes [BCDA-3260](https://jira.cms.gov/browse/BCDA-3260)

Removed the flag to switch between HICN and MBI for beneficiary identification as MBI is the only identifier being used currently

### Proposed Changes

Removed references to HICN from the code

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Deployed to https://bcda-ci.adhocteam.us/job/BCDA%20-%20Build%20and%20Package/1694/

### CODE REVIEW

File: bcda/client/bluebutton.go [hunk 1]
```diff
@@ -112,11 +112,7 @@ func (bbc *BlueButtonClient) GetPatient(patientID, jobID, cmsID, since string, t
 
 func (bbc *BlueButtonClient) GetPatientByIdentifierHash(hashedIdentifier, patientIdMode string) (string, error) {
```
Comment (msnook) [inline bcda/client/bluebutton.go:113]: is `patientIdMode` necessary now?

File: bcda/client/bluebutton.go [hunk 2]
```diff
@@ -112,11 +112,7 @@ func (bbc *BlueButtonClient) GetPatient(patientID, jobID, cmsID, since string, t
...
 	params := GetDefaultParams()
-
-	identifier := "hicn-hash"
-	if patientIdMode == "MBI_MODE" {
-		identifier = "mbi-hash"
-	}
+	identifier := "mbi-hash"
```
Comment (msnook) [inline bcda/client/bluebutton.go:115]: Is this necessary?  Can it just be baked-in at line 118?

**Comment by msnook** [comment]
Should `PATIENT_IDENTIFIER_MODE` be removed from the docker-compose configs?

File: bcda/models/models_test.go [hunk 3]
```diff
@@ -793,7 +766,7 @@ func (s *ModelsTestSuite) TestGetBlueButtonID_CCLFBeneficiary() {
 
 	// Should be making two calls to BB for the MBI_MODE attemptsm, but this number will be four with the earlier test in this method.
 	// This is due to the fact that we are not relying on cached identifiers
-	bbc.AssertNumberOfCalls(s.T(), "GetPatientByIdentifierHash", 4)
+	bbc.AssertNumberOfCalls(s.T(), "GetPatientByIdentifierHash", 2)
 
-	os.Unsetenv("PATIENT_IDENTIFIER_MODE")
+	// os.Unsetenv("PATIENT_IDENTIFIER_MODE")
```
Comment (mtrang1263) [inline bcda/models/models_test.go:771]: Can we remove this instead of commenting it out?

File: bcdaworker/main_test.go [hunk 4]
```diff
@@ -268,7 +268,7 @@ func TestWriteEOBDataToFile_BlueButtonIDNotFound(t *testing.T) {
 	defer db.Delete(&cclfFile)
 
 	bbc := testUtils.BlueButtonClient{}
-	bbc.On("GetPatientByIdentifierHash", mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return("", errors.New("No beneficiary found for HICN"))
+	bbc.On("GetPatientByIdentifierHash", mock.AnythingOfType("string")).Return("", errors.New("No beneficiary found for HICN"))
```
Comment (msnook) [inline bcdaworker/main_test.go:271]: Should `HICN` be in the error here?

**Pull request merged**